src/worker/worker-pool.js

```diff
@@ -9,7 +9,8 @@ class WorkerPool {
 
     offload(fn, args) {
```

Comment:
we should use async await when possible. Function should be async with no Promise returned